CMake fails to "Configure" on Windows 10 when using SDL2-2.0.14

The exact error message reads:

CMake Error at D:/programs/cmake-3.19.5-win64-x64/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:218 (message):
  Could NOT find SDL2W: Found unsuitable version "", but required is at least
  "2.0.4" (found 1)

I am attaching a screenshot from the CMake GUI.

I tried to use the following tools for compilation:
- g++ compiler ver. 8.1.0 (contained in MinGW-W64)
- CMake 3.19.5 for Windows
- SDL2 2.0.14 development library

![BStone_SDL2_error_2021-02-17](https://user-images.githubusercontent.com/6830489/108206132-46799080-7126-11eb-9e5b-7cebae34b964.JPG)

New screenshot as per comment:

![BStone_SDL2_error_2021-02-17_02](https://user-images.githubusercontent.com/6830489/108256998-7f355c00-715e-11eb-95d7-9d92f5670776.JPG)

Screenshot from the CLI version of CMake as per comment:

![BStone_SDL2_error_2021-02-17_03](https://user-images.githubusercontent.com/6830489/108263785-fb33a200-7166-11eb-8362-4113b979d0bf.JPG)

Screenshot from VS Code execution of CMake as per comment:

![BStone_SDL2_error_2021-02-17_04](https://user-images.githubusercontent.com/6830489/108263844-0b4b8180-7167-11eb-9923-710a462777e3.JPG)

--- a/src/cmake/FindSDL2W.cmake
+++ b/src/cmake/FindSDL2W.cmake
@@ -1,9 +1,8 @@
 #[[
 
-CMake wrapper for SDL2 module.
+SDL2W - CMake wrapper for SDL2.
 
-
-Copyright (c) 2018-2020 Boris I. Bendovsky (bibendovsky@hotmail.com) and Contributors.
+Copyright (c) 2018-2021 Boris I. Bendovsky (bibendovsky@hotmail.com) and Contributors.
 
 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
@@ -41,7 +40,7 @@ Targets:
 
 cmake_minimum_required (VERSION 3.1.3 FATAL_ERROR)
 
-set (SDL2W_VERSION "1.0.3")
+set (SDL2W_VERSION "1.0.4")
 message (STATUS "[SDL2W] Version: ${SDL2W_VERSION}")
 
 set (SDL2W_TMP_TARGET "${CMAKE_FIND_PACKAGE_NAME}::${CMAKE_FIND_PACKAGE_NAME}")
@@ -297,16 +296,22 @@ endforeach ()
 foreach (SDL2W_TMP_INCLUDE_DIR IN LISTS SDL2W_TMP_SDL2_INC_DIRS)
 	set (SDL2W_TMP_SDL_VERSION_H ${SDL2W_TMP_INCLUDE_DIR}/SDL_version.h)
 
-	# Convert POSIX path to Windows one (MinGW)
+	# [MSYS2] Convert POSIX path to Windows one.
 	#
 	if (MINGW)
+		unset (SDL2W_TMP_SDL_VERSION_H_MSYS2)
+
 		execute_process (
 			COMMAND "sh" "-c" "cmd //c echo ${SDL2W_TMP_SDL_VERSION_H}"
 			TIMEOUT 7
-			OUTPUT_VARIABLE SDL2W_TMP_SDL_VERSION_H
+			OUTPUT_VARIABLE SDL2W_TMP_SDL_VERSION_H_MSYS2
 			ERROR_QUIET
 			OUTPUT_STRIP_TRAILING_WHITESPACE
 		)
+
+		if (SDL2W_TMP_SDL_VERSION_H_MSYS2 MATCHES "SDL_version\\.h$")
+			set (SDL2W_TMP_SDL_VERSION_H ${SDL2W_TMP_SDL_VERSION_H_MSYS2})
+		endif ()
 	endif ()
 
 	# Extract version.
@@ -395,7 +400,11 @@ foreach (SDL2W_TMP_INCLUDE_DIR IN LISTS SDL2W_TMP_SDL2_INC_DIRS)
 	endif ()
 endforeach ()
 
-message (STATUS "[SDL2W] Found SDL version: ${SDL2W_TMP_VERSION_STRING}")
+if (SDL2W_TMP_VERSION_STRING)
+	message (STATUS "[SDL2W] Found SDL version: ${SDL2W_TMP_VERSION_STRING}")
+else ()
+	message (FATAL_ERROR "[SDL2W] Failed to detect SDL version.")
+endif ()
 
 
 # Default handler.

